### PR TITLE
User api

### PR DESCRIPTION
**Pull request by saurabhprasun20**

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java [hunk 1]
```diff
@@ -53,4 +53,52 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
...
+    public UserGetDTO getUserByUserName(@RequestHeader("X-Auth-Token") String token, @PathVariable Long userId){
+        UserGetDTO userGetDTO = new UserGetDTO();
+        return  userGetDTO;
+
+    }
+
+    @GetMapping("/user/login")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @ResponseBody
+    public UserGetDTO login(@RequestBody UserPostDTO userPostDTO){
```
Comment (sedited) [inline src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java]: Can you implement a UserAuthDTO that is returned here, containing just the Token?

Comment (saurabhprasun20) [reply src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java]: Done.

File: src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java [hunk 2]
```diff
@@ -53,4 +54,52 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
...
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public void logout(@RequestHeader("X-Auth-Token") String token,@PathVariable Long userId){
+        return;
+    }
+
+    @PutMapping("/user/{userId}/edit")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public UserGetDTO updateUser(@RequestHeader("X-Auth-Token") String token,@RequestBody User user, @PathVariable Long userId){
```
Comment (sedited) [inline src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java]: You're using the UserGetDTO object here again for the put request. This is a bit confusing. Can you either implement another object, e.g. UserPutDTO, or use a more generic name if you want to re-use them?

Comment (saurabhprasun20) [reply src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java]: Changed the return type to UserUpdateDTO

File: src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java [hunk 3]
```diff
@@ -53,4 +55,52 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
...
+
+    @PutMapping("/user/{userId}/password")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public void updatePassword(@RequestHeader("X-Auth-Token") String token,@RequestBody UserPostDTO userPostDTO){
+        return;
+    }
+
+    @PutMapping("/user/{userId}/invitation")
+    public List<Long> invitation(@RequestHeader("X-Auth-Token") String token,@PathVariable Long userId){
```
Comment (sedited) [inline src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java]: The request body is missing here, in the class diagram I am using a UserPutDTO, but up to you, if you want to use a separate one again. There also no need to return anything here, but be sure to set the response status to OK.

File: src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java [hunk 4]
```diff
@@ -53,4 +55,52 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
...
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public void logout(@RequestHeader("X-Auth-Token") String token,@PathVariable Long userId){
+        return;
+    }
+
+    @PutMapping("/user/{userId}/edit")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public UserUpdateDTO updateUser(@RequestHeader("X-Auth-Token") String token, @RequestBody User user, @PathVariable Long userId){
```
Comment (sedited) [inline src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java]: The RequestBody should be a DTO object, just create one if it does not fit the class diagram scheme for now, we can update it later again.

File: src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java [hunk 5]
```diff
@@ -53,4 +55,52 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
...
+        UserUpdateDTO userUpdateDTO = new UserUpdateDTO();
+        return userUpdateDTO;
+    }
+
+    @PutMapping("/user/{userId}/password")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public void updatePassword(@RequestHeader("X-Auth-Token") String token,@RequestBody UserPostDTO userPostDTO){
```
Comment (sedited) [inline src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java]: I'm also not happy with the UserPostDTO here. I'll tweak the class diagram according to your choice.

Comment (saurabhprasun20) [reply src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java]: Not implementing password change functionality. API not needed

File: src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java (hunk 4 above)
Comment (saurabhprasun20) [reply src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java]: Changed to UpdateUserDTO

**Review by sedited — APPROVED** [review]

**Pull request merged**